monitoring/tracing/tracer.go

```diff
@@ -4,20 +4,24 @@ package tracing
 
 import (
 	"errors"
+	"time"
 
-	"contrib.go.opencensus.io/exporter/jaeger"
 	prysmTrace "github.com/prysmaticlabs/prysm/v5/monitoring/tracing/trace"
-	"github.com/prysmaticlabs/prysm/v5/runtime/version"
 	"github.com/sirupsen/logrus"
-	"go.opencensus.io/trace"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
 )
 
 var log = logrus.WithField("prefix", "tracing")
 
-// Setup creates and initializes a new tracing configuration..
+// Setup creates and initializes a new Jaegar tracing configuration with opentelemetry.
 func Setup(serviceName, processName, endpoint string, sampleFraction float64, enable bool) error {
 	if !enable {
-		trace.ApplyConfig(trace.Config{DefaultSampler: trace.NeverSample()})
+		// If tracing is disabled, return immediately
```

Comment:
do we not need to reset like otel.SetTracerProvider(tp) or something in this case? or no

Reply:
yeah, i think you might need to 

```suggestion
		otel.SetTracingProvider(noop.NewTracerProvider())
```

See: https://pkg.go.dev/go.opentelemetry.io/otel/trace@v1.30.0/noop#NewTracerProvider